src/hooks/useStore.ts

```diff
@@ -0,0 +1,47 @@
+import { useContext, useRef, useSyncExternalStore, useCallback } from 'react';
+import { type Store } from '../utils/storeManager';
+
+type StoreSelector<T, U> = (state: T) => U;
+
+/**
+ * A generic hook for accessing and updating store state
+ * @param StoreContext
+ * @param selector
+ * @param initialState
+ */
+export function useStore<T, U>(
+  StoreContext: React.Context<Store<T> | null>,
+  selector: StoreSelector<T, U>,
+  initialState: T,
+) {
+  const store = useContext(StoreContext);
+  if (!store) {
+    throw new Error('useStore must be used within a StoreProvider');
+  }
+  // Ensure the stability of the selector function using useRef
+  const selectorRef = useRef(selector);
+  selectorRef.current = selector;
+  /**
+   * useSyncExternalStore - a new API introduced in React18
+   * useSyncExternalStore simply tracks changes in an external store that is not dependent on React
+   * through useState and useEffect
+   * and helps with re-rendering and state sync through the setter of useState
+   */
+  const state = useSyncExternalStore(
```

Comment:
It's due to our peerDependencies; we can't use `useSyncExternalStore` 🥲

```json
  "peerDependencies": {
    "react": "^16.8.6 || ^17.0.0 || ^18.0.0",
    "react-dom": "^16.8.6 || ^17.0.0 || ^18.0.0"
  },
```

Reply:
👉 https://www.npmjs.com/package/use-sync-external-store

Reply:
@bang9 Oh thank you for the catch! Just updated to use the 3rd party hook https://github.com/sendbird/sendbird-uikit-react/pull/1216/commits/f74c5713c50e17255eed0bc815b9d413196e2cae